deepchecks/nlp/checks/data_integrity/frequent_substrings.py

```diff
@@ -0,0 +1,227 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains Frequent Substrings check."""
+import typing as t
+from collections import defaultdict
+from typing import Dict
+
+import pandas as pd
+
+from deepchecks.core import CheckResult, ConditionCategory, ConditionResult
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.nlp import Context, SingleDatasetCheck
+from deepchecks.nlp._shared_docs import docstrings
+from deepchecks.nlp.text_data import TextData
+from deepchecks.utils.strings import format_percent
+
+__all__ = ['FrequentSubstrings']
+
+
+@docstrings
+class FrequentSubstrings(SingleDatasetCheck):
+    """Checks for frequent substrings in the dataset.
+
+    Parameters
+    ----------
+    n_to_show : int, default: 5
+        number of most common duplicated samples to show.
+    n_samples : int, default: 10_000_000
+        number of samples to use for this check.
+    random_state : int, default: 42
+        random seed for all check internals.
+    min_threshold: float, default: 0.05
+        minimum threshold for tagging a substring as frequent
+    min_ngram: int, default: 4
+        minimum n for the n-gram extraction
+    epsilon: float, default: 0.02
+        maximum difference for filtering long substrings
+    significant_threshold: float, default: 0.3
+        all the samples above the frequency would be returned
+
+    """
+
+    def __init__(
+        self,
+        n_to_show: int = 5,
+        n_samples: int = 10_000_000,
+        random_state: int = 42,
+        min_threshold: float = 0.05,
+        min_ngram: int = 4,
+        epsilon=0.02,
+        significant_threshold: float = 0.3,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.n_to_show = n_to_show
+        self.n_samples = n_samples
+        self.random_state = random_state
+        self.min_threshold = min_threshold
+        self.min_ngram = min_ngram
+        self.epsilon = epsilon
+        self.significant_threshold = significant_threshold
+
+    @staticmethod
+    def _extract_ngrams(text, n):
+        words = text.split()
+        return [' '.join(words[i:i + n]) for i in range(len(words) - n + 1)]
+
+    @staticmethod
+    def _get_peak_cut_ind(df, min_diff=0.05):
+        df['diff'] = df['Frequency'].diff()
+        df[df['diff'] <= min_diff]['diff'] = 0
```

Comment:
I'm considering whether the diff should be on on the frequency (like you did here) or rather on the change in frequency  - meaning `(freq_i - freq_i_+1) / freq_i`. What do you think?